RequestService/RequestService.Repo/Repository.cs

```diff
@@ -1602,36 +1602,36 @@ public async Task<bool> UpdateAllJobStatusToOpenForRequestAsync(int requestId, i
             }
         }
 
-        public async Task<List<int>> UpdateRequestStatusToCancelledAsync(int requestId, int createdByUserID, CancellationToken cancellationToken)
+        public async Task<List<int>> UpdateRequestStatusAsync(JobStatuses jobStatus, int requestId, int createdByUserID, CancellationToken cancellationToken)
         {
             List<int> result = new List<int>();
 
-            byte cancelledJobStatus = (byte)JobStatuses.Cancelled;
+            byte byteJobStatus = (byte)jobStatus;
 
-            var jobs = _context.Job.Where(w => w.RequestId == requestId && w.JobStatusId != cancelledJobStatus);
+            var jobs = _context.Job.Where(w => w.RequestId == requestId && w.JobStatusId != byteJobStatus);
```

Comment:
We should exclude any jobs which are Cancelled, i.e. update anything which is != jobStatus && != cancelled